File: CodeSharpDoc/CSharpCodeStructureAnalyser/Models/StructSummariesInfos.cs
namespace CSharpCodeStructureAnalyser.Models;

using System.Collections.Generic;
using System.Text.Json.Serialization;

public class MethodSummaryInfo
{
    [JsonPropertyName("code_start_index")]
    public int CodeStartIndex { get; set; }

    [JsonPropertyName("summary")]
    public string Summary { get; set; }

    public MethodSummaryInfo(int codeStartIndex, string summary = "")
    {
        CodeStartIndex = codeStartIndex;
        Summary = summary ?? "";
    }
}

public class StructSummariesInfos
{
    [JsonPropertyName("file_path")]
    public string FilePath { get; set; }

    [JsonPropertyName("index_shift_code")]
    public int IndexShiftCode { get; set; }

    [JsonPropertyName("indent_level")]
    public int IndentLevel { get; set; }

    [JsonPropertyName("summary")]
    public string Summary { get; set; }

    [JsonPropertyName("methods")]
    public List<MethodSummaryInfo> Methods { get; set; }

    public StructSummariesInfos(string filePath, int indexShiftCode, int indentLevel, string summary, List<MethodSummaryInfo>? methods = null)
    {
        FilePath = filePath;
        IndexShiftCode = indexShiftCode;
        IndentLevel = indentLevel;
        Summary = summary;
        Methods = methods ?? new List<MethodSummaryInfo>();
    }
}

File: CodeDoc/CodeDoc.API/src/models/struct_summaries_infos.py
class MethodSummaryInfo:
    def __init__(self, code_start_index: int, summary: str = None):
        self.code_start_index: int = code_start_index
        self.summary: str = summary
    
    def to_dict(self):
        return {
            'code_start_index': self.code_start_index,
            'summary': self.summary
        }

class StructSummariesInfos:
    def __init__(self, file_path: str, index_shift_code: int, indent_level: int, summary: str = None , methods: list[MethodSummaryInfo] = []):
        self.file_path: str = file_path
        self.index_shift_code: int = index_shift_code
        self.indent_level: int = indent_level
        self.summary: str = summary
        self.methods: list[MethodSummaryInfo] = methods

    def to_dict(self):
        return {
            'file_path': self.file_path,
            'index_shift_code': self.index_shift_code,
            'indent_level': self.indent_level,
            'summary': self.summary,
            'methods': [method.to_dict() for method in self.methods]
        }
